pyfr/plugins/tavg.py

```diff
@@ -118,23 +122,12 @@ def _eval_acc_exprs(self, intg):
             # Prepare the substitutions dictionary
             subs = dict(zip(pnames, psolns))
 
-            # Compute any required gradients
+            # Prepare any required gradient
             if self._gradpnames:
-                # Gradient operator and J^-T matrix
-                gradop, rcpjact = self._gradop[i], self._rcpjact[i]
-                nupts = gradop.shape[1]
-
+                gradps = grads_eles[j]
                 for pname in self._gradpnames:
-                    psoln = subs[pname]
-
-                    # Compute the transformed gradient
-                    tgradpn = gradop @ psoln
-                    tgradpn = tgradpn.reshape(self.ndims, nupts, -1)
-
-                    # Untransform this to get the physical gradient
-                    gradpn = np.einsum('ijkl,jkl->ikl', rcpjact, tgradpn)
-                    gradpn = gradpn.reshape(self.ndims, nupts, -1)
-
+                    idx = self.elementscls.privarmap[self.ndims].index(pname)
```

Comment:
We might want to pre-compute these indices and store them in self._gradpnames (suitably renamed, of course).  This should enable us to simplify these two loops.

Reply:
`self_gradpidxs`?

Reply:
Something like that, where the variable itself is a list of tuples (name, idx).

Reply:
Done